maven-core/src/main/java/org/apache/maven/ReactorReader.java

```diff
@@ -266,7 +266,12 @@ private boolean isPackagedArtifactUpToDate( MavenProject project, File packagedA
                 long outputFileLastModified = Files.getLastModifiedTime( outputFile ).toMillis();
                 if ( outputFileLastModified > artifactLastModified )
                 {
-                    logger.warn( "Packaged artifact is not up-to-date compared to the build output directory" );
+                    LOGGER.warn(
+                            "Packaged artifact for {} is not up-to-date compared to the build output directory; "
```

Comment:
One other thing I was missing is: which one is used?
Could this be compressed to something like: 

    For ARTIFACTID the jar is older than o.a.m.some.File.class: using target/classes.

Reply:
Since this PR is already merged, I've created a follow-up where we can discuss the actual message: https://github.com/apache/maven/pull/416.